src/Select.js

```diff
@@ -102,7 +102,7 @@ class Select extends React.Component {
 			this.setState({ required: false });
 		}
 
-		if (this.state.inputValue && this.props.value !== nextProps.value) {
+		if (this.state.inputValue && this.props.value !== nextProps.value && this.props.onSelectResetsInput) {
```

Comment:
I think better `if (this.state.inputValue && this.props.value !== nextProps.value && nextProps.onSelectResetsInput) {`

Reply:
Hi, @JedWatson 
Sorry I didn't see the merge. Can you please also add this change?
Thank you.